src/coreComponents/meshUtilities/InternalWellboreGenerator.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file InternalWellboreGenerator.cpp
+ */
+
+#include "InternalWellboreGenerator.hpp"
+#include "managers/DomainPartition.hpp"
+#include "mpiCommunications/PartitionBase.hpp"
+
+namespace geosx
+{
+using namespace dataRepository;
+
+InternalWellboreGenerator::InternalWellboreGenerator( string const & name, Group * const parent ):
+  InternalMeshGenerator( name, parent )
+{}
+
+REGISTER_CATALOG_ENTRY( MeshGeneratorBase, InternalWellboreGenerator, string const &, Group * const )
```

Comment:
I assume that you have a new class only for registering the product in the factory, am I right?
I cannot see this "name" in your xml entry.
Another possibility is to inherit the specialized features in classes and keep the generic in a base class.

Reply:
Yes I will add features specialized for wellbore class then I will update the xml file using the registered name.